internal/collections/named.go

```diff
@@ -101,11 +101,41 @@ type NamedCollectionNames struct {
 }
 
 func (c *NamedCollectionNames) FindRegex(key *regexp.Regexp) []types.MatchData {
-	panic("selection operator not supported")
+	var res []types.MatchData
+
+	for k, data := range c.collection.Map.data {
+		if key.MatchString(k) {
```

Comment:
Please use early termination to reduce the indentation.

Reply:
This pattern is also used [here](https://github.com/blotus/coraza/blob/e1d83b287b699c2b5b5293f535489e88278e2e7c/internal/collections/map.go#L61) for example, should it be also updated or do you want to keep this for another PR ?